Training/for_loop.cpp

```diff
@@ -1,50 +1,43 @@
 #include <iostream>
 #include "for_loop.h"
 
+//Es- Consider a switch statement:
```

Comment:
I understand that some of these comments are orthogonal to hackerrank, so they're mainly notes or considerations. I'm not grading your work!

In this case, _efficiency_ is going to be tricky given the lack of complexity. **In general**, cascading `if-else` statements are bad, because they introduce a lot of hypothetical branching (which may or may not get compiled away), whereas an array access definitely does not introduce branching.

Functions oftentimes get inlined automatically during compilation and, depending on the way cache is laid out, popping on a stack frame can even be beneficial sometimes. Simple switches with enumerated cases will actually compile out very efficiently-- using array-like memory access to jump to the case label immediately, instead of stepping through each condition.

Poorly formed switches can't do that and will compile out to a series of if-else branches, so it's a buyer's beware sort of situation.

Reply:
Understood.